File: Repeated String/repeatedString.py
def repeatedString(s, n):
    dictionary = {'a': 0}

    length = n // len(s)

    if 'a' not in s:
        return 0

    for i in s:
        if i == 'a':
            dictionary['a'] += 1
    
    remaining = n - len(s) * length

    total = int(dictionary['a'] * length)

    if remaining > 0:
        for i in range(remaining):
            if s[i] == 'a':
                total += 1

    return total














# def repeatedString(s, n):
#     dictionary = {}

#     length = n // len(s)

#     if 'a' not in s:
#         return 0

#     for i in s:
#         if i == 'a':
#             if 'a' not in dictionary:
#                 dictionary['a'] = 1
#             else:
#                 dictionary['a'] += 1
    
#     remaining = n - len(s) * length

#     total = int(dictionary['a'] * length)

#     if remaining > 0:
#         for i in range(remaining):
#             if s[i] == 'a':
#                 total += 1

#     return total

File: Repeated String/repeatedString.js
function repeatedString(s, n) {
    const incl = s.includes('a')

    if (!incl) {
        return 0
    }

    let dictionary = {'a': 0}
    
    const length = Math.floor(n / s.length)

    for (let i = 0; i < s.length; i++) {
        if (s[i] === 'a') {
            dictionary['a'] += 1
        }
    }

    let remaining = n - s.length * length

    let total = dictionary['a'] * length

    if (remaining > 0) {
        for (let i = 0; i < remaining; i++) {
            if (s[i] === 'a') {
                total += 1
            }
        }
    }

    return total
}